src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/Int64Converter.cs

```diff
@@ -7,12 +7,26 @@ internal sealed class Int64Converter : JsonConverter<long>
     {
         public override long Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
+            if (reader.TokenType == JsonTokenType.String &&
+                options != null &&
+                (JsonNumberHandling.AllowReadingFromString & options.NumberHandling) != 0)
+            {
+                return reader.GetInt64WithQuotes();
+            }
+
             return reader.GetInt64();
         }
 
         public override void Write(Utf8JsonWriter writer, long value, JsonSerializerOptions options)
         {
-            writer.WriteNumberValue(value);
+            if (options != null && ((JsonNumberHandling.WriteAsString & options.NumberHandling) != 0))
```

Comment:
This is going to collide with this PR: https://github.com/dotnet/runtime/pull/39323

Reply:
As a merge conflict anyway... I can hold off on that if you expect this to go in before snap.

Reply:
Go for it.. I'm hoping to make it before the snap, but still need to fix the netfx failure (hopefully just this)/